src/storage/src/hummock/sstable/mod.rs

```diff
@@ -155,19 +154,6 @@ impl Sstable {
         !self.filter_reader.is_empty()
     }
 
-    pub fn may_match(&self, dist_key: &[u8]) -> bool {
```

Comment:
how can we handle the fail_point `disable_bloom_filter` if we remove this branch

Reply:
> how can we handle the fail_point `disable_bloom_filter` if we remove this branch

this fail point has been ineffective for a long time 😢 
at least since #7492

Reply:
Can we also remove the failpoint test at this PR ?

Reply:
> Can we also remove the failpoint test at this PR ?

fixed